### PR TITLE
#159316712: create and host app

### PR DESCRIPTION
**Pull request by tutugodfrey**
**#### What does this PR do?
- Add configurations to host app on heroku

#### Description of Task to be completed?
- create heroku pipeline
- create heroku apps for staging and production
- update default app.json
- add Procfile and script for starting the application

#### How should this be manually tested?
- heroku pipeline will generate a review app on everypull request
- the current state of the application can be tested from the generated review app

#### What are the relevant pivotal tracker stories?
#159316712

#### Any background context you want to provide?
view app on https://authorshaven-staging-pr-20.herokuapp.com/api

#### Screenshots (if appropriate)
image

image

**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,14 +3,20 @@
...
+  },
   "scripts": {
-    "start": "babel-node index.js --exec babel-node",
+    "start": "babel-node index.js",
     "dev": "NODE_ENV=development nodemon ./index.js --exec babel-node",
-    "lint": "eslint .",
     "pretest": "npm run posttest && sequelize --env test db:migrate && sequelize --env test db:seed:all",
     "test": "cross-env NODE_ENV=test nyc mocha --compilers js:babel-core/register ./test/**/*.js",
     "posttest": "sequelize --env test db:migrate:undo:all",
-    "coverage": "npm run pretest && cross-env NODE_ENV=test babel-node ./node_modules/babel-istanbul/lib/cli.js cover ./node_modules/mocha/bin/_mocha && npm run posttest"
```
Comment (megame24) [inline package.json]: reverse your coverage to this script to enable your travis build to pass.

Comment (megame24) [inline package.json]: reverse your coverage to this script to enable your travis build to pass.

Comment (tutugodfrey) [reply package.json]: ok

**Comment by megame24** [comment]
Sorry, I deleted your comment: `ok` to my suggestion to reverse the `coverage` script back to how it was to enable the travis build pass.

**Review by megame24 — APPROVED** [review]

File: models/User.js [hunk 2]
```diff
@@ -40,14 +40,14 @@ export default (sequelize, DataTypes) => {
   }, {
     hooks: {
       beforeCreate: (user) => {
-        user.salt = crypto.randomBytes(16).toString('hex');
-        user.hash = crypto // eslint-disable-next-line no-param-reassign
-          .pbkdf2Sync(user.hash, user.salt, 10000, 512, 'sha512')
+        user.salt = crypto.randomBytes(16).toString('hex'); // eslint-disable-line no-param-reassign
+        user.hash = crypto // eslint-disable-line no-param-reassign
+          .pbkdf2Sync(user.hash, user.salt, 10000, 512, 'sha512') // eslint-disable-line no-param-reassign
```
[the remark this answers is not in the document]
Comment (Fattylee) [reply models/User.js:49]: Nice work. Just pay more attention to houndci max length error

**Pull request closed without merging**